From 148115fcc40f1651a5d15fa34c9a8c528e7147bb Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sun, 4 Aug 2019 21:00:33 +0200
Subject: [PATCH] OGRExpatRealloc(): fix double-free when size to allocate is
 above the default 10MB threshold. Fixes
 https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=16178. Credit to OSS
 Fuzz

---
 gdal/ogr/ogr_expat.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gdal/ogr/ogr_expat.cpp b/gdal/ogr/ogr_expat.cpp
index 49e2c73a13b..0d6c4a9ba7e 100644
--- a/gdal/ogr/ogr_expat.cpp
+++ b/gdal/ogr/ogr_expat.cpp
@@ -89,7 +89,6 @@ static void* OGRExpatRealloc( void *ptr, size_t size )
     if( CanAlloc(size) )
         return realloc(ptr, size);
 
-    free(ptr);
     return nullptr;
 }
 
